### PR TITLE
Fix for attn_softmax_bf16 when generation_config is None

### PR DESCRIPTION
**Pull request by schoi-habana**
This addresses an error from the attn_softmax_bf16 change in cases where self.model.generation_config exists but is 'None'.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/habana/transformers/trainer.py [hunk 1]
```diff
@@ -853,7 +853,7 @@ def hpu_deepspeed_checkpointing(function, *checkpoint_args):
                     self.control = self.callback_handler.on_step_begin(args, self.state, self.control)
 
                 # attn_softmax_bf16 is enabled only for llama
-                if hasattr(self.model, "generation_config"):
+                if hasattr(self.model, "generation_config") and self.model.generation_config is not None:
```
Comment (regisss) [inline optimum/habana/transformers/trainer.py:856]: Not sure we need to add this. If `self.model.generation_config.attn_softmax_bf16` is None, the condition `self.model.generation_config.attn_softmax_bf16` in the if-statement right below will not be True and thus nothing is done no?

Comment (schoi-habana) [reply optimum/habana/transformers/trainer.py:856]: We met an error saying:  'NoneType' object has no attribute 'attn_softmax_bf16' so had to check if self.model.generation_config is not None

Comment (regisss) [reply optimum/habana/transformers/trainer.py:856]: Ah my bad I read the code too quickly!
How did you get that error? What was the command or use case?

Comment (mandy-li) [reply optimum/habana/transformers/trainer.py:856]: We got the error why FT llama2-7b

**Review by mandy-li — APPROVED** [review]

File: optimum/habana/transformers/trainer.py (hunk 1 above)
Comment (schoi-habana) [reply optimum/habana/transformers/trainer.py:856]: for sequence classification

**Pull request merged**